component/http/v2/encoding/json/json.go

```diff
@@ -0,0 +1,125 @@
+// Package json contains helper methods to handler requests and responses more easily.
+package json
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/beatlabs/patron/encoding"
+	"github.com/beatlabs/patron/encoding/json"
+)
+
+const (
+	singleStar = "*"
+	doubleStar = "*/*"
+)
+
+// ReadRequest validates the request headers and decodes into the provided payload.
+func ReadRequest(req *http.Request, payload interface{}) error {
+	err := validatingContentType(req)
+	if err != nil {
+		return err
+	}
+
+	return json.Decode(req.Body, payload)
+}
+
+func validatingContentType(req *http.Request) error {
```

Comment:
Naming suggestion: 
1. validate instead of validating
2. Header suffix
```suggestion
func validateContentTypeHeader(req *http.Request) error {
```